src/test/java/org/apache/sysds/test/functions/builtin/BuiltinGaussianClassifierTest.java

```diff
@@ -0,0 +1,142 @@
+		testGaussianClassifier(80, 30, 0.3, 5);
+	}
+
+	@Test
+	public void testSmallSparseTenClasses() {
+		testGaussianClassifier(80, 30, 0.3, 10);
+	}
+
+	public void testGaussianClassifier(int rows, int cols, double sparsity, int classes)
+	{
+		loadTestConfiguration(getTestConfiguration(TEST_NAME));
+		String HOME = SCRIPT_DIR + TEST_DIR;
+		fullDMLScriptName = HOME + TEST_NAME + ".dml";
+		;
+		double varSmoothing = 1e-9;
+
+		List<String> proArgs = new ArrayList<>();
+		proArgs.add("-args");
+		proArgs.add(input("X"));
+		proArgs.add(input("Y"));
+		proArgs.add(String.valueOf(varSmoothing));
+		proArgs.add(output("priors"));
+		proArgs.add(output("means"));
+		proArgs.add(output("determinants"));
+		proArgs.add(output("invcovs"));
+
+		programArgs = proArgs.toArray(new String[proArgs.size()]);
+
+		rCmd = getRCmd(inputDir(), Double.toString(varSmoothing), expectedDir());
+		
+		double[][] X = getRandomMatrix(rows, cols, 0, 100, sparsity, -1);
+		double[][] Y = getRandomMatrix(rows, 1, 0, 1, 1, -1);
+		for(int i=0; i<rows; i++){
+			Y[i][0] = (int)(Y[i][0]*classes) + 1;
+			Y[i][0] = (Y[i][0] > classes) ? classes : Y[i][0];
+		}
+
+		writeInputMatrixWithMTD("X", X, true);
+		writeInputMatrixWithMTD("Y", Y, true);
+
+		runTest(true, EXCEPTION_NOT_EXPECTED, null, -1);
+
+		runRScript(true);
+
+		HashMap<CellIndex, Double> priorR = readRMatrixFromExpectedDir("priors");
+		HashMap<CellIndex, Double> priorSYSTEMDS= readDMLMatrixFromOutputDir("priors");
+		HashMap<CellIndex, Double> meansRtemp = readRMatrixFromExpectedDir("means");
+		HashMap<CellIndex, Double> meansSYSTEMDStemp = readDMLMatrixFromOutputDir("means");
+		HashMap<CellIndex, Double> determinantsRtemp = readRMatrixFromExpectedDir("determinants");
+		HashMap<CellIndex, Double> determinantsSYSTEMDStemp = readDMLMatrixFromOutputDir("determinants");
+		HashMap<CellIndex, Double> invcovsRtemp = readRMatrixFromExpectedDir("invcovs");
+		HashMap<CellIndex, Double> invcovsSYSTEMDStemp = readDMLMatrixFromOutputDir("invcovs");
+
+		double[][] meansR = TestUtils.convertHashMapToDoubleArray(meansRtemp);
+		double[][] meansSYSTEMDS = TestUtils.convertHashMapToDoubleArray(meansSYSTEMDStemp);
+		double[][] determinantsR = TestUtils.convertHashMapToDoubleArray(determinantsRtemp);
+		double[][] determinantsSYSTEMDS = TestUtils.convertHashMapToDoubleArray(determinantsSYSTEMDStemp);
+		double[][] invcovsR = TestUtils.convertHashMapToDoubleArray(invcovsRtemp);
+		double[][] invcovsSYSTEMDS = TestUtils.convertHashMapToDoubleArray(invcovsSYSTEMDStemp);
+
+		TestUtils.compareMatrices(priorR, priorSYSTEMDS, Math.pow(10, -5.0), "priorR", "priorSYSTEMDS");
+		TestUtils.compareMatricesBitAvgDistance(meansR, meansSYSTEMDS, 5L,5L, this.toString());
+		TestUtils.compareMatricesBitAvgDistance(determinantsR, determinantsSYSTEMDS, (long)2E+12,(long)2E+12, this.toString());
+		TestUtils.compareMatricesBitAvgDistance(invcovsR, invcovsSYSTEMDS, (long)2E+20,(long)2E+20, this.toString());
```

Comment:
i would probably add the execution of the inverse multiplied with the result,
just as you stated in the pr it gives the original.

if you add this to the tests, would it still make sense to run the R script?

Reply:
thanks for the feedback. I will commit the suggested changes as soon as possible.
Just one follow up question to this specific suggestion:
Do you mean I should check wether covariance * invCovariance is indeed the identity matrix?
If so, since the covariance matrix is not returned (but just the inverse), should I compute, 
the respective covariance matrices again in the test dml file, and then return the product, to check
wether it is indeed the identity? 
If this is not what you meant, it would be great if you could clarify. 
Thanks :)

Reply:
yes covariance * invCovariance == Identity.

and if there are other interesting properties. My initial understanding was that there was a way to reconstruct the input from the output, but i also might have missed a point, and looking a covariance there does not seem to be a way of doing that?

Reply:
reconstructing the input from the output is afaik not really possible - since you lose for example the number of input samples provided to the algorithm. However computing the covariance matrix again, and multiplying it with its inverse is definitely a way of proving that the inverse is indeed correct, although its R aquivalent seems to be kinda different (but as already mentioned, just because of floating point differences).

Reply:
For now I have not added such a test, since the cov matrix itself is not part of the output parameters. The only way to perform such a test is to calculate the cov matrices again inside the dml test file, and then multiply them with their respective inverses. Of course I can implement it like this, but I just wanted to make sure if this is the correct approach to do, since it seems a bit weird to me to recalculate the cov matrices inside the test files.